poh/benches/poh.rs

```diff
@@ -65,3 +76,129 @@ fn bench_poh_lock_time_per_batch(bencher: &mut Bencher) {
         poh.hash(DEFAULT_HASHES_PER_BATCH);
     })
 }
+
+#[bench]
+fn bench_poh_recorder_record_transaction_index(bencher: &mut Bencher) {
+    let ledger_path = get_tmp_ledger_path_auto_delete!();
+    let blockstore =
+        Blockstore::open(ledger_path.path()).expect("Expected to be able to open database ledger");
+    let GenesisConfigInfo { genesis_config, .. } = create_genesis_config(2);
+    let bank = Arc::new(Bank::new_for_tests(&genesis_config));
+    let prev_hash = bank.last_blockhash();
+
+    let (mut poh_recorder, _entry_receiver, _record_receiver) = PohRecorder::new(
+        0,
+        prev_hash,
+        bank.clone(),
+        Some((4, 4)),
+        bank.ticks_per_slot(),
+        Arc::new(blockstore),
+        &std::sync::Arc::new(LeaderScheduleCache::new_from_bank(&bank)),
+        &PohConfig::default(),
+        Arc::new(AtomicBool::default()),
+    );
+    let h1 = hash(b"hello Agave, hello Anza!");
+    poh_recorder.set_bank_with_transaction_index_for_test(bank.clone());
+    poh_recorder.tick();
+    assert_eq!(
+        poh_recorder
+            .working_bank
+            .as_ref()
+            .unwrap()
+            .transaction_index
+            .unwrap(),
+        0
+    );
+
+    bencher.iter(|| {
+        let tx0 = test_tx();
+        let tx1 = test_tx();
+
+        let _record_result = poh_recorder
+            .record(bank.slot(), h1, vec![tx0.into(), tx1.into()])
+            .unwrap()
+            .unwrap();
+        assert!(
+            poh_recorder
+                .working_bank
+                .as_ref()
+                .unwrap()
+                .transaction_index
+                .unwrap()
+                > 0
+        );
+        let tx = test_tx();
+        let h2 = hash(b"foobar");
+        let record_result = poh_recorder
+            .record(bank.slot(), h2, vec![tx.into()])
+            .unwrap()
+            .unwrap();
+        assert!(record_result > 0);
+        poh_recorder
+            .working_bank
+            .as_ref()
+            .unwrap()
+            .transaction_index
+            .unwrap();
+        assert!(
+            poh_recorder
+                .working_bank
+                .as_ref()
+                .unwrap()
+                .transaction_index
+                .unwrap()
+                > 0
+        );
```

Comment:
already asserting on the result above; they should be the same so I don't think we really need to assert on both, just 1 sanity check that the index is non-zero